File: src/pocketcloud/cloud/command/impl/HelpCommand.php
<?php

namespace pocketcloud\cloud\command\impl;

use pocketcloud\cloud\command\argument\def\StringArgument;
use pocketcloud\cloud\command\Command;
use pocketcloud\cloud\command\CommandManager;
use pocketcloud\cloud\command\sender\ICommandSender;

final class HelpCommand extends Command {

    public function __construct() {
        parent::__construct("help", "List all commands");
        $this->addParameter(new StringArgument(
            "command",
            true
        ));
    }

    public function run(ICommandSender $sender, string $label, array $args): bool {
        $command = $args["command"] ?? null;
        $commands = $command === null ? CommandManager::getInstance()->getAll() : (($tmp = CommandManager::getInstance()->get($command)) === null ? CommandManager::getInstance()->getAll() : [$tmp]);

        foreach ($commands as $command) {
            $sender->info("§b" . $command->getName() . " §8- §r" . $command->getDescription() . " §8- §b" . $command->getUsage());
        }
        return true;
    }
}

File: src/main/java/de/pocketcloud/cloud/console/command/impl/HelpCommand.java
package de.pocketcloud.cloud.console.command.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.parameter.def.LiteralCommandParameter;
import de.pocketcloud.cloud.console.command.sender.CommandSender;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@CommandDescription(name = "help", description = "List all commands", aliases = {"?"})
public final class HelpCommand extends Command {

    @Override
    public void prepare() {
        addParameter(new LiteralCommandParameter("command", true, "Command not found"));
    }

    @Override
    public boolean run(CommandSender sender, String label, Map<String, Object> args, Map<String, Object> flags) {
        List<Command> commands = new ArrayList<>();
        if (args.containsKey("command")) {
            commands.add((Command) args.get("command"));
        } else {
            commands.addAll(PocketCloud.getInstance().commandManager().getAll());
        }

        sender.info("Commands §8(§b{}§8)§r:", commands.size());
        for (Command command : commands) {
            if (command.getAliases().length > 0) {
                sender.info("§b{} §8- §r{} §8- §r[§c{}§r]", command.getName(), command.getDescription(), String.join("§8, §c", command.getAliases()));
            } else {
                sender.info("§b{} §8- §r{} §8- §r[]", command.getName(), command.getDescription());
            }
        }

        return true;
    }
}